crates/autopilot/src/decoded_settlement.rs

```diff
@@ -365,12 +362,30 @@ impl DecodedSettlement {
 pub struct Fees {
     /// The UID of the order associated with these fees.
     pub order: OrderUid,
+    /// The type of fee that was executed.
+    pub kind: FeeKind,
     /// The executed fees in the sell token.
     pub sell: U256,
     /// The executed fees in the native token.
     pub native: U256,
 }
 
+impl Fees {
+    /// Get the surplus fee for this order.
+    pub fn executed_surplus_fee(&self) -> Option<U256> {
+        match self.kind {
+            FeeKind::User => None,
```

Comment:
nit: The only caller of this function does `.unwrap_or_default()` so I think we might as well already do that here.

Reply:
Fixed.